components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -5392,15 +5430,126 @@ private int addSubscription(Connection connection, ApiTypeWrapper apiTypeWrapper
         return subscriptionId;
     }
 
+    /**
+     * Returns whether a given API Name already exists
+     *
+     * @param apiName        Name of the API
+     * @param organization Identifier of an Organization
+     * @return true/false
+     * @throws APIManagementException if failed to get API Names
+     */
+    public List<String> getAPIVersionsMatchingApiNameAndOrganization(String apiName, String organization)
+            throws APIManagementException {
+
+        List<String> versionList = new ArrayList<String>();
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement ps = connection
+                     .prepareStatement(SQLConstants.GET_VERSIONS_MATCHES_API_NAME_AND_ORGANIZATION_SQL)) {
+            boolean initialAutoCommit = connection.getAutoCommit();
+            ResultSet resultSet = null;
```

Comment:
use try with resources for PreparedStatement and ResultSet.

Please do this in all places below.